plugins/woocommerce/src/Internal/Admin/ProductForm/FormFactory.php

```diff
@@ -0,0 +1,246 @@
+	 * @return Card|WP_Error New section or WP_Error.
+	 */
+	public static function add_section( $id, $plugin_id, $args ) {
+		$new_section = self::create_item( 'section', 'Section', $id, $plugin_id, $args );
+		if ( is_wp_error( $new_section ) ) {
+			return $new_section;
+		}
+		self::$form_sections[ $id ] = $new_section;
+		return $new_section;
+	}
+
+	/**
+	 * Returns form config.
+	 *
+	 * @return array form config.
+	 */
+	public static function get_form_config() {
+		return array(
+			'fields'   => self::get_fields(),
+			'cards'    => self::get_cards(),
+			'sections' => self::get_sections(),
+		);
+	}
+
+	/**
+	 * Returns list of registered fields.
+	 *
+	 * @param array $sort_by key and order to sort by.
+	 * @return array list of registered fields.
+	 */
+	public static function get_fields( $sort_by = array(
+		'key'   => 'order',
+		'order' => 'asc',
+	) ) {
+		return self::get_items( 'field', 'Field', $sort_by );
+	}
+
+	/**
+	 * Returns list of registered cards.
+	 *
+	 * @param array $sort_by key and order to sort by.
+	 * @return array list of registered cards.
+	 */
+	public static function get_subsections( $sort_by = array(
+		'key'   => 'order',
+		'order' => 'asc',
+	) ) {
+		return self::get_items( 'subsection', 'Subsection', $sort_by );
+	}
+
+	/**
+	 * Returns list of registered sections.
+	 *
+	 * @param array $sort_by key and order to sort by.
+	 * @return array list of registered sections.
+	 */
+	public static function get_sections( $sort_by = array(
+		'key'   => 'order',
+		'order' => 'asc',
+	) ) {
+		return self::get_items( 'section', 'Section', $sort_by );
+	}
+
+	/**
+	 * Returns list of registered items.
+	 *
+	 * @param string       $type Form component type.
+	 * @param class-string $class_name Class of component type.
+	 * @param array        $sort_by key and order to sort by.
+	 * @return array       list of registered items.
+	 */
+	private static function get_items( $type, $class_name, $sort_by = array(
+		'key'   => 'order',
+		'order' => 'asc',
+	) ) {
+		$item_list = self::${ 'form_' . $type . 's' };
+		$class     = 'Automattic\\WooCommerce\\Internal\\Admin\\ProductForm\\' . $class_name;
+		$items     = array_values( $item_list );
+		if ( class_exists( $class ) && method_exists( $class, 'sort' ) ) {
+			usort(
+				$items,
+				function ( $a, $b ) use ( $sort_by, $class ) {
+					return $class::sort( $a, $b, $sort_by );
+				}
+			);
+		}
+		return $items;
+	}
+
+	/**
+	 * Creates a new item.
+	 *
+	 * @param string       $type Form component type.
+	 * @param class-string $class_name Class of component type.
+	 * @param string       $id Item id.
+	 * @param string       $plugin_id Plugin id.
+	 * @param array        $args additional arguments for item.
+	 * @return Field|Card|Section|WP_Error New product form item or WP_Error.
+	 */
+	private static function create_item( $type, $class_name, $id, $plugin_id, $args ) {
+		$item_list = self::${ 'form_' . $type . 's' };
```

Comment:
Totally optional, but this looks like it could be as tid bit fragile if there was ever a name change/etc here. Could do a mapping or something similar, but not required at all.